vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java

```diff
@@ -987,4 +1013,74 @@ private String generateHtmlValue() {
         }
     }
 
+    private class AsDelta
+            implements HasValue<ValueChangeEvent<String>, String> {
+        /**
+         * Sets the value of this editor in the
+         * <a href="https://github.com/quilljs/delta">Quill Delta</a> format. If
+         * the new value is not equal to {@code getValue()}, fires a value
+         * change event. Throws {@code NullPointerException}, if the value is
+         * null.
+         * <p>
+         * Note: {@link Binder} will take care of the {@code null} conversion
+         * when integrates with the editor, as long as no new converter is
+         * defined.
+         *
+         * @param value
+         *            the new value in Delta format, not {@code null}
+         */
+        @Override
+        public void setValue(String value) {
+            RichTextEditor.this.setValue(value);
+        }
+
+        /**
+         * Returns the current value of this editor in the
+         * <a href="https://github.com/quilljs/delta">Quill Delta</a> format. By
+         * default, the empty editor will return an empty string.
+         *
+         * @return the current value.
+         */
+        @Override
+        public String getValue() {
+            return RichTextEditor.this.getValue();
+        }
+
+        @Override
+        public Registration addValueChangeListener(
+                ValueChangeListener<? super ValueChangeEvent<String>> valueChangeListener) {
+            return RichTextEditor.this
+                    .addValueChangeListener(valueChangeListener);
+        }
+
+        @Override
+        public void setReadOnly(boolean b) {
```

Comment:
Similar methods in `AsHtml` have JavaDocs. Should we also add them to the methods in `AsDelta` or remove them from the `AsHtml` counterparts?

Reply:
I don't think it's necessary, as the methods inherit docs from `HasValue`, and the most interesting methods in this class from a developer perspective are the getter and setter for the value. Most of the other methods are only necessary to get Binder working correctly.